lib/core/output_sink/es_sink.rb

```diff
@@ -26,7 +26,7 @@ def initialize(index_name, request_pipeline, flush_threshold = 50)
     def ingest(document)
       return if document.blank?
 
-      @operation_queue << { :index => { :_index => index_name, :_id => document[:id], :data => document } }
+      @operation_queue << { :index => { :_index => index_name, :_id => document['id'], :data => document } }
```

Comment:
Here it still uses `:id`:

https://github.com/elastic/connectors-ruby/blob/ba2b87a3676dee416da48bbdec55cc8f06d03bd9/lib/core/sync_job_runner.rb#L63-L64

It depends on connector implementation in what's inside of `document`, I think it's reasonable to add some validation in the framework level (i.e. connector service)

Reply:
Great catch, gonna fix it here too!

Reply:
I *think* with the fix I put in it's not needed to fix it in that place anymore. I've checked locally and don't see any dupes now.